pkg/rancher-desktop/config/settingsImpl.ts

```diff
@@ -32,19 +32,28 @@ let settings: Settings | undefined;
  */
 function loadFromDisk(): Settings {
   // Throw an ENOENT error if the file doesn't exist; the caller should know what to do.
-  const rawdata = fs.readFileSync(join(paths.config, 'settings.json'));
-  const cfg = clone(defaultSettings);
+  const settingsPath = join(paths.config, 'settings.json');
+  const rawdata = fs.readFileSync(settingsPath);
+  const defaultConfig = clone(defaultSettings);
 
   try {
     // If the existing settings file is partial, fill in the missing fields with defaults.
-    merge(cfg, JSON.parse(rawdata.toString()));
+    const currentConfig = JSON.parse(rawdata.toString());
+
+    if (!('version' in currentConfig)) {
+      throw new SettingsError(`No version specified in ${ settingsPath }`);
+    }
+    merge(defaultConfig, currentConfig);
```

Comment:
`currentConfig` here is possibly some old version; if we merge the current-version defaults in first that seems like an issue. We should migrate the (complete?) settings before merging with the defaults.